[BE] Propose candidates by technology to invite

nspc97 opened the pull request:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Added route /user/list filtered by technologies & sorted by seniority

Commit:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Added route /user/list filtered by technologies & sorted by seniority

MihaiDruta commented on src/Controller/UserController.php:
```diff
@@ -396,4 +398,81 @@ public function userChangePassword(
     }
+
+    /**
+     * Change password of User.
```
Wrong description...

Commit:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Added route /user/list filtered by technologies & sorted by seniority

SpartakusMd commented on src/Controller/UserController.php:
```diff
@@ -396,4 +398,81 @@ public function userChangePassword(
+        $userListFilter->setFilterFields((array)$filter);
+
+        $sorting = $request->query->get('sortBy');
+        $userListSort->setSortingFields((array)$sorting);
```
Pagination is missing. In order to have the possibility to extract all the items in one page we should handle the case when `per_page=-1`. If it is exactly `-1` => return all the items.

Commit:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Added route /user/list filtered by technologies & sorted by seniority

SpartakusMd commented on src/Repository/UserRepository.php:
```diff
@@ -68,4 +70,27 @@ public function getApplicantsForActivity(ApplicantsListSort $applicantsListSort,
+        $queryBuilder = $this->createQueryBuilder('user');
+        $queryBuilder
+            ->select('user');
+        if ($userListFilter->technology !== null) {
```
Use `if (!empty($userListFilter->technology))`. If someone will do a request `...?filter[technology]=&...` (note the missing value for technology) you will get a database error `unexpected closing parenthesis in "technology IN ()"`

Commit:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Added route /user/list filtered by technologies & sorted by seniority

Commit:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Added pagination

Commit:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Fixed variables name

SpartakusMd commented on src/Handlers/UserHandler.php:
```diff
@@ -0,0 +1,73 @@
+        UserListFilter $userListFilter
+    ): array {
+
+        if ($userListPagination->pageSize === -1) {
```
This check should be done in `getPaginatedUserList()`.

Commit:
[BE] Propose candidates by technology to invite
https://trello.com/c/cSl69V4Z/53-be-propose-candidates-by-technology-to-invite

Pagination response fields are now correct for per_page=-1 (listing all users)
